_posts/2020-11-30-developer-survey-2020.md

```diff
@@ -0,0 +1,99 @@
+---
+layout: post
+title: Solita Developer Survey 2020 - what technologies do we use and like?
+author: juholei
+excerpt: >
+  todo
+tags:
+ - Software Development
+
+---
+
+Developer survey has now become a tradition here at Solita. Every two years we survey our developer-minded people regarding what technologies and practices we use in our every-day work. The first survey was conducted in 2016 and you can find a blog post about those results in [our previous blog post](https://dev.solita.fi/java/2016/05/13/Developers-love-spaces.html) and then in 2018 [(blog post in Finnish)](https://www.solita.fi/blogit/development-teknologiakartoitus/). This year we once again took a loot at the survey and came up with a total of 27 questions that would give us interesting insight into what we use, what we like and what could be big in the future. Comparing these results to the previous ones can also give us some information on trends - what technologies are increasing in use and what are in decline?
+
+## Languages we use
+
+![All languages](/img/developer-survey-2020/all_languages.png)
+
+The first question in the survey regarding programming languages asked to list all the languages you have worked with in the previous year. JavaScript and SQL were the dominant ones: SQL is present in some form in most projects we do and JavaScript is ubiquitous in frontend and hard to miss even if you primarily use different languages for frontend. Following them were Java and Python.
+
+Apart from languages that are used for the actual system, scripting languages like Sh/Bash (third most used overall), PowerShell and Groovy were widely used.
+
+The following questions about main backend and frontend languages were new this year and were added to get better information on how the languages are used. As such, these can't be compared as such to previous years. Usage comparisons are then based on the question about all languages used.
+
+![Backend languages](/img/developer-survey-2020/backend_languages.png)
+
+Moving on to more focused questions, Java proved to be the most used language for backend development. 45% reported Java being one of the primary backend languages in ther projects. Rounding out top 3, are then Clojure with about 19% usage and Kotlin with 18%. Compared to results from 2018, percentage of Kotlin users rose from 15% to 24% (comparison based on the question of all languages used). Java and Clojure usage were both a little bit down when comparing 2020 to 2018, which could be explained by the increased usage of Clojure. Neither seem to be going away though, as shown by being top 1 and 2 used languages for backend and both are well liked (more on that later).
```

Comment:
> Java and Clojure usage were both a little bit down when comparing 2020 to 2018, which could be explained by the increased usage of Clojure.

Should this be "increased usage of Kotlin"? Seems weird if Clojure is down because of it's increased usage :)